web/wp-content/plugins/ix-header-bidder/class-ix-header-bidder.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * IX Header Bidder
+ *
+ *  @package     IX Header Bidder
+ *  Plugin Name: IX Header Bidder.
+ *  Version: 1.0.0 .
+ *  Description: IX Header Bidder. <a href='https://readersdigest.atlassian.net/browse/WPDT-3432' target='_blank'>Read more at WPDT-3432</a>.
+ *  Author: DRDT Team.
+ *  Plugin URI: https://readersdigest.atlassian.net/browse/WPDT-3432.
+ *  Text Domain: ix-header-bidder.
+ */
+
+/**
+ * Including file for the settings.
+ *
+ * @file
+ */
+require 'inc/class-ix-settings.php';
+
+/**
+ *  Class Header Bidder.
+ */
+class IX_Header_Bidder {
+	/**
+	 * This will have the value of the script
+	 *
+	 * @var String
+	 */
+	private $script_url = '';
+	const VERSION       = '1.0.0';
+
+	/**
+	 *  Constructor.
+	 */
+	public function __construct() {
+		$ix_settings      = new IX_Settings();
+		$this->script_url = $ix_settings->script_url;
+		if ( $this->script_url ) {
+			add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
```

Comment:
Using `$this` here breaks `remove_action` API. Additionally, adding hooks in constructors makes classes difficult to test.
If classes are needed, a better choice would be using static functions for hooks, having a public `init` function, and then initialize the plugin with something like `add_action( 'init', array( 'IX_Header_Bidder', 'init' ) );` - however, in this case it seems classes are used for namespacing only.